ql/src/java/org/apache/hadoop/hive/ql/io/IOContext.java

```diff
@@ -187,6 +188,14 @@ public void parseRecordIdentifier(Configuration configuration) {
     }
   }
 
+  public void parsePositionDeleteInfo(Configuration configuration) {
+    this.pdi = PositionDeleteInfo.parseFromConf(configuration);
```

Comment:
Would it worth to set the `pdi` fields one-by-one instead of creating a new object for every row?